### DISCUSSION TITLE
Interactive JSON

### DISCUSSION THREAD

**Discussion by spence-novata** [deref/exo#544] [Ideas]
I would love to see the GUI provide a Chrome dev tools type experience. Specifically for JSON objects dumped to log I'd love to be able to expand large objects in a collapsing tree and pop it into a variable that I can explore with a JS console.

I'm stoked that I found exo as I had the itch to build this myself so super glad you are tackling this problem.

**Comment by brandonbloom** [comment]
Analyzing output for JSON objects is a lovely idea, thanks! We do a tiny bit of this sort of thing for links: https://github.com/deref/exo/blob/ece7542288f2835c2585dc3f741169358e3feefd/gui/src/components/logs/FormattedLogMessage.stories.svelte#L28-L34 - Would really like to handle a wide range of entity extraction extensions for the log viewer. Also, the under-development overhaul branch (PR #543) has a much richer event system, so we should be able to support integration with structured logging in the future as well.

Will have to give some more thought to the JS console idea...
